core/src/main/java/org/neo4j/ogm/context/GraphEntityMapper.java

```diff
@@ -160,6 +155,53 @@ public GraphEntityMapper(MetaData metaData, MappingContext mappingContext) {
         return results;
     }
 
+    /**
+     * Executes method annotated with @PostLoad on node and relationship entities with given ids
+     *
+     * This method must be executed after processing all result rows:
+     * - post load must be called after the entity is fully hydrated
+     * - to avoid executing post load multiple times on same entity
+     *
+     * @param nodeIds nodeIds
+     * @param edgeIds edgeIds
+     */
+    public void executePostLoad(Set<Long> nodeIds, Set<Long> edgeIds) {
+        Set<Long> seenIds = new HashSet<>();
+        for (Long id : nodeIds) {
+            Object o = mappingContext.getNodeEntity(id);
+
+            if (!seenIds.contains(id)) {
+                executePostLoad(o);
+                seenIds.add(id);
+            }
+        }
+
+        seenIds.clear();
+        for (Long id : edgeIds) {
+            Object o = mappingContext.getRelationshipEntity(id);
+
+            if (!seenIds.contains(id)) {
+                if (o != null) {
+                    executePostLoad(o);
+                }
+                seenIds.add(id);
+            }
+
+        }
+    }
+    private void executePostLoad(Object instance) {
+        ClassInfo classInfo = metadata.classInfo(instance);
+        MethodInfo postLoadMethod = classInfo.postLoadMethodOrNull();
+        if (postLoadMethod != null) {
+            final Method method = classInfo.getMethod(postLoadMethod);
+            try {
+                method.invoke(instance);
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                logger.error("Failed to execute post load method", e);
```

Comment:
When the PostLoad method does not respect the "contract" (public, no arg method), low level java exceptions are thrown like `IllegalArgumentException: wrong number of arguments`
Maybe some guidance could be useful like "Cannot call PostLoad annotated method {method fqn}. Make sure it is public and has no arguments"

Reply:
Changed, I have also changed it to warn as it is not a blocking error.